web/src/services/posthogClient.ts

```diff
@@ -0,0 +1,43 @@
+import { PostHog } from "posthog-node";
+import getConfig from "next/config";
+const { publicRuntimeConfig } = getConfig();
+interface CaptureEventParams {
+  event: string;
+  distinctId: string;
+  properties: object;
+}
+
+// Function to initialize PostHog
+function initializePostHog(): PostHog | null {
+  if (!publicRuntimeConfig.NEXT_PUBLIC_POSTHOG_API_KEY) {
+    // API Key not available
+    return null;
+  }
+
+  return new PostHog(publicRuntimeConfig.NEXT_PUBLIC_POSTHOG_API_KEY, {
+    host:
+      publicRuntimeConfig.NEXT_PUBLIC_POSTHOG_HOST || "https://app.posthog.com",
+  });
+}
+
+// Initialize PostHog only if it hasn't been initialized yet
+let posthogClient: PostHog | null = null;
+
+export function captureEvent({
+  event,
+  distinctId,
+  properties,
+}: CaptureEventParams): void {
+  if (!posthogClient) {
+    posthogClient = initializePostHog();
+  }
+  if (posthogClient) {
+    posthogClient.capture({
```

Comment:
important to add the geoip_disable prop to make sure we don't use any server location. also worth syncing with @igorosip0v for advice on `flushAt` (or check PostHog docs)

Reply:
@andy-t-wang we can add some `flush` function with `await posthog.flush()` inside to use it on the server where we need to make sure that we sent all events before leaving endpoint, for example.

Alternatively, we can use `await posthog.shutdownAsync()` inside `captureEvent`. But this will stop the client (and send all hanging events). And everytime we will use `captureEvent` we will create a client and kill it at the end.

Reply:
Or we can set the `flushAt: 1` property in the `PostHog` options, it should flush every event, but maybe it will be more optimized if we use `flush` manually. Not sure here